app/assets/javascripts/component_guide/application.js

```diff
@@ -1,2 +1,4 @@
+//= require_tree ../govuk_publishing_components/vendor/polyfills
+//= require govuk/vendor/polyfills/Element/prototype/closest.js
```

Comment:
This needs to be at the top of `step-by-step-nav.js` otherwise it will only be available in the component guide and not in the apps importing `step-by-step-nav`.

We also need `//= require govuk/vendor/polyfills/Element/prototype/classList.js ` as we're using it in a few places and it's not supported in IE9 and below and has partial support in 10 and 11.

Reply:
I didn't put it in `step-by-step-nav.js` because I didn't want to introduce duplication if other components went a similar way, but that's okay because of gzip?

`classList.js` is interesting - you're right, it isn't supported, but I had no errors relating to that during testing. Is it possible that the govuk-frontend polyfill is already included, which is why it was working? I guess it would be sensible to add it again though, particularly if duplication is handled?

Reply:
Yes, we benefit from `govuk-frontend` polyfills in the component guide as we're requiring all components (hence all the polyfills used in `govuk-frontend`). The duplication is not handled especially since `govuk-frontend` code already has the polyfills attached to each component. While not happy with this I am keen to carry on with this model until we move to ES6 modules, have a way to bundle our JavaScripts and benefit from tree-shaking.

Reply:
I think I'm following you, so I've moved the polyfill calls into the component itself.

Oddly, I couldn't get the `govuk-frontend` polyfill for `closest` to work. Tried it in IE8 in XP and Windows 7 to no avail. It fell over in `setActiveStepClass` at line `302`. I've re-added back in another polyfill, which seems to work. We obviously don't want the duplication, but I'm not sure where it's going wrong - the `classList` polyfill seems to be working fine.

Reply:
🤔 hmm.. I'll try to have a look at the `closest` polyfill from `govuk-frontend`